toolchain/base/mem_usage.h

```diff
@@ -0,0 +1,113 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_BASE_MEM_USAGE_H_
+#define CARBON_TOOLCHAIN_BASE_MEM_USAGE_H_
+
+#include <cstdint>
+
+#include "common/map.h"
+#include "common/set.h"
+#include "llvm/ADT/STLFunctionalExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "toolchain/base/yaml.h"
+
+namespace Carbon {
+
+// Types supporting memory usage tracking should implement:
+//
+//   // Collects memory usage of members.
+//   auto CollectMemUsage(llvm::StringRef label, AddMemUsageFn add_mem_usage)
+//       const -> void;
+//
+// The label should be concatenated with any child labels using MemUsageLabel in
```

Comment:
"MemUsageLabel" seems like it should be a function or something from this file, but this is the only mention of this word. Do you mean `MemUsage::ConcatLabel`?